pkg/cmd/template/regular_input.go

```diff
@@ -71,11 +71,18 @@ func (s *RegularFilesSource) Output(out TemplateOutput) error {
 		return out.Err
 	}
 
+	nonYamlFileNames := []string{}
 	switch {
 	case len(s.opts.outputDir) > 0:
 		return files.NewOutputDirectory(s.opts.outputDir, out.Files, s.ui).Write()
 	case len(s.opts.outputFiles) > 0:
 		return files.NewOutputDirectory(s.opts.outputFiles, out.Files, s.ui).WriteFiles()
+	default:
+		for _, file := range out.Files {
```

Comment:
this looks much cleaner compared to previous iteration. i think we are on the right path!